lib/git_helper/local_code.rb

```diff
@@ -73,24 +81,8 @@ def branch
       `git branch`.scan(/\*\s([\S]*)/).first.first
     end
 
-    def default_branch(project_name, external_client, client_type)
-      if client_type == :octokit # GitHub repository
-        external_client.repository(project_name).default_branch
-      elsif client_type == :gitlab # GitLab project
-        page_number = 1
-        counter = 1
-        branches = []
-
-        while counter > 0
-          break if default_branch = branches.select { |branch| branch.default }.first
-          page_branches = external_client.branches(project_name, page: page_number, per_page: 100)
-          branches = page_branches
-          counter = page_branches.count
-          page_number += 1
-        end
-
-        default_branch.name
-      end
+    def default_branch
+      `git symbolic-ref refs/remotes/origin/HEAD | sed "s@^refs/remotes/origin/@@" | tr -d "\n"`
```

Comment:
Gather the default branch from the symbolic refs locally, not the CLI. Although this is less accurate, it doesn't require the octokit client or gitlab client, and I think it'd be rare where they wouldn't match. It doesn't assume, too... it asks if the command is correct.